File: LeetCode/1304.uniquesum.py
# Given an integer n, return any array containing n unique integers such that they add up to 0.

 

# Example 1:

# Input: n = 5
# Output: [-7,-1,1,3,4]
# Explanation: These arrays also are accepted [-5,-1,1,2,3] , [-3,-1,2,-2,4].

# Example 2:

# Input: n = 3
# Output: [-1,0,1]

# Example 3:

# Input: n = 1
# Output: [0]

 

# Constraints:

#     1 <= n <= 1000

class Solution:
    def sumZero(self, n: int) -> List[int]:
        # arr = []
        # j = 0
        # for i in range(n//2):
        #     j+=1
        #     arr.append(j)
        #     arr.append(-1*j)
        # if n%2 != 0:
        #     arr.append(0)
        # return arr
        return(range(1-n,n,2))

File: LeetCode/cpp/1304.uniquesum.cpp
class Solution {
public:
    vector<int> sumZero(int n) {
        int i;
        vector<int> res;
        for(i = 1 ; i < n/2 + 1; i++){
            res.push_back(i);
            res.push_back(-i);
        }
        if( n % 2 == 1){
            res.push_back(0);
        }
        return res;
    }
};